request/src/main/java/com/vimeo/networking2/VimeoApiClient.kt

```diff
@@ -422,6 +410,147 @@ interface VimeoApiClient {
      */
     fun deleteConnectedApp(type: ConnectedAppType, callback: VimeoCallback<Unit>): VimeoRequest
 
+    /**
+     * Create a folder that will be used to organize videos.
+     *
+     * @param uri The URI of the user's projects connection.
+     * @param name The name of the folder.
+     * @param privacy The privacy of the folder.
+     * @param callback The callback which will be notified of the request completion.
+     *
+     * @return A [VimeoRequest] object to cancel API requests.
+     */
+    fun createFolder(
+        uri: String,
+        name: String,
+        privacy: FolderViewPrivacyType,
+        callback: VimeoCallback<Folder>
+    ): VimeoRequest
+
+    /**
+     * Create a folder that will be used to organize videos.
+     *
+     * @param user The user whose projects connection will be used for the request.
+     * @param name The name of the folder.
+     * @param privacy The privacy of the folder.
+     * @param callback The callback which will be notified of the request completion.
+     *
+     * @return A [VimeoRequest] object to cancel API requests.
+     */
+    fun createFolder(
+        user: User,
+        name: String,
+        privacy: FolderViewPrivacyType,
+        callback: VimeoCallback<Folder>
+    ): VimeoRequest
+
+    /**
+     * Delete a folder.
+     *
+     * @param folder The folder being deleted, URI should not be null or empty.
```

Comment:
`URI should not be null or empty` is the standard warning we apply to functions in the client that accept a DTO instead of a URI. It's a good idea to remove it though, as you suggested, since we gracefully respond with a failure if the URI is null.